### ISSUE TITLE
Magento2 resizes and recompresses original image during upload

### ISSUE THREAD

**Issue by php4umagento** [magento/magento2#17910]
### Preconditions

1. 2.2.5 and below versions are affected
2.

### Steps to reproduce

1. Create jpg file larger than 1900x1200 pixels
2. Login to admin
3. Create or update existing product image by uploading new file you created
4. Observe that after upload image is resized to not exceed 1900x1200
5. File is also processed by php using compression quality setting

### Expected result

1. File is uploaded as it is - it is not modified by Magento 2

### Actual result

1. File is resized
2. File is recompressed

There are 3 problems with the current solution
1. Max width and height are hardcoded as class constants - \Magento\Backend\Block\Media\Uploader
2. Compression quality is a class variable, but it is not exposed in admin / config - \Magento\Catalog\Model\Product\Image
3. Templates in admin are using class constants not static methods or class methods

I propose to create a config in admin under Advanced > Developer > Image processing
with 3 new items - compression quality, max width, max height

**Comment by magento-engcom-team** [comment]
Hi @php4umagento. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@php4umagento do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by php4umagento** [comment]
@magento-engcom-team give me 2.2.5 instance

**Comment by magento-engcom-team** [comment]
Hi @php4umagento. Thank you for your request. I'm working on Magento 2.2.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @php4umagento, here is your [Magento instance](https://i-17910-2-2-5.engcom.dev.magento.com).
Admin access: [https://i-17910-2-2-5.engcom.dev.magento.com/admin](https://i-17910-2-2-5.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by php4umagento** [comment]
Product SKU Test - uploaded 4k image - it was resized to 1920x918px

**Comment by ghost** [comment]
@php4umagento, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by eduard13** [comment]
Hi @php4umagento , looks like this is a duplicate of https://github.com/magento/magento2/issues/13747, and a such feature has been already added in 2.2.
Thank you.

**Comment by mahsn** [comment]
#webjump

**Comment by magento-engcom-team** [comment]
@mahsn thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @mahsn. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `G1 Passed` will be added to the issue automatically. Please, edit issue description if needed, until label `G1 Passed` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `G2 Passed` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `acknowledged` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report is acknowledged.

**Comment by mahsn** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @mahsn. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @mahsn, here is your [Magento instance](https://i-17910-2-3-develop.engcom.dev.magento.com).
Admin access: [https://i-17910-2-3-develop.engcom.dev.magento.com/admin](https://i-17910-2-3-develop.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by jtomka** [comment]
#mlau

**Comment by magento-engcom-team** [comment]
@jtomka thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @jtomka. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by jtomka** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @jtomka. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @jtomka, here is your [Magento instance](https://i-17910-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-17910-2-2-develop.instances.magento-community.engineering/admin](https://i-17910-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by jtomka** [comment]
Tested in 2.2-develop and 2.3-develop and the issue is not reproducible any more.

The 3 proposed new items - compression quality, max width, max height - are present in the backend (Stores / Configuration / System / Images Upload Configuration) and taken into account during image upload, when Enable Frontend Resize is set to Yes.

When Enable Frontend Resize is set to No, image is uploaded as is, without resizing.

**Comment by BB-000** [comment]
Stores / Configuration / Advanced / System / Images Upload Configuration

Is this planned to be available at store configuration level?

**Comment by BB-000** [comment]
Is this planned to be available at store configuration level?

**Comment by BB-000** [comment]
I have 'Enable Frontend Resize' turned off, but my image is still increasing in file size. It is a .png, increases by about 40kb
I have cleared all caches including _pub/media/catalog/product/cache_
🤷‍♀️
